File: unit4/Q5.java
/*Create a class with a method that prints "This is parent class" and its subclass with another 
method that prints "This is child class". Now, create an object for each of the class and 
call
● method of parent class by object of parent class
● method of child class by object of child class
● method of parent class by object of child class*/

class Parent {
  void showP() {
    System.out.println("Method of parent class called.");
  }
}

class Child extends Parent {
  void showC(){
    System.out.println("Method of child class called.");
  }
}

public class Q5 {
  public static void main(String args[]) {
    // creating parent object and calling parent function
    Parent p = new Parent();
    p.showP();

    // creating child object
    Child c = new Child();
    // calling child function
    c.showC();
    // calling parent function
    c.showP();
  }
}

File: unit4/python codes/Q5.py
# Create a class with a method that prints "This is parent class" and its subclass with another 
# method that prints "This is child class". Now, create an object for each of the class and 
# call
# ● method of parent class by object of parent class
# ● method of child class by object of child class
# ● method of parent class by object of child class

class Parent:
  def showP(self):
    print("Parent method is called")

class Child(Parent):
  def showC(self):
    print("Child method is called")

if __name__ == "__main__":
  print()
  
  # Creating a parent object
  p = Parent()
  p.showP() # calling parent method

  # Creating child object
  c = Child()
  c.showC() # calling child process
  c.showP() # calling parent process using child object